src/algebra/direct_sum/ring.lean

```diff
@@ -224,6 +224,20 @@ begin
     exact of_eq_of_graded_monoid_eq (pow_succ (graded_monoid.mk _ a) n).symm, },
 end
 
+lemma of_list_dprod {α} (l : list α) (fι : α → ι) (fA : Π a, A (fι a)) :
+  of A _ (l.dprod fι fA) = (l.map $ λ a, of A (fι a) (fA a)).prod :=
+begin
+  induction l,
+  { simp only [list.map_nil, list.prod_nil, list.dprod_nil],
+    refl },
+  { simp only [list.map_cons, list.prod_cons, list.dprod_cons, ←l_ih, direct_sum.of_mul_of],
+    refl },
```

Comment:
Adding `dsimp only [list.dprod_index_cons],` changes the goal to be of the form `x = x`, but I still need `refl` to close it here. Interesting `simp` is not able to apply `dprod_index_cons`, only `dsimp` is.